site/src/workers/schedule.worker.ts

```diff
@@ -0,0 +1,56 @@
+import type { WasmContext } from "@/quacs-rs";
+
+// @ts-expect-error: after initialization, this won't be null
+let ctx: WasmContext = null;
+
+async function waitForInitialization(): Promise<void> {
+  while (
+    ctx === null ||
+    // @ts-expect-error: For some reason, the wasm context is able to be constructed before
+    // it's fully initialized... We need to check for that here by digging into its internals.
+    ctx.ptr === 0
```

Comment:
Race condition with the `ctx = new wasm.WasmContext();` line? Not sure if there's a way to make assignments atomic or use mutexes or something.

Reply:
Yeah, I'm not sure what causes that.  I know that the ptr will be set to null if the rust code takes ownership, though after fixing an issue with that (in the code removed from the previous comment) the error still occurred occasionally - not sure if there's an easy fix or I missed something, but an extra equality check isn't the end of the world.

Reply:
> I know that the ptr will be set to null if the rust code takes ownership
Maybe Rust in one thread tries to take ownership while another thread owns it?

Reply:
That could cause it, but there should only be one thread with Rust here... maybe while trying to track down the bug I accidentally set up a second web worker?  That could explain some weird behavior I was seeing in the console.

Reply:
I guess we didn't have this issue before because the global `RwLock`s  prevented the race conditions? It's probably fine for now though.